legal-api/tests/unit/resources/v2/test_business.py

```diff
@@ -296,30 +309,65 @@ def test_get_business_with_incomplete_info(
     jwt,
     test_name,
     legal_type,
+    owner_legal_type,
     identifier,
+    owner_identifier,
     has_missing_business_info,
     missing_business_info_warning_expected,
 ):
     """Assert that SP/GPs with missing business info is populating warnings list."""
     with nested_session(session):
         if has_missing_business_info:
-            legal_entity = factory_legal_entity(entity_type=legal_type, identifier=identifier)
+            if owner_legal_type:
+                legal_entity = factory_legal_entity(entity_type=owner_legal_type, identifier=owner_identifier)
+            else:
+                legal_entity = factory_legal_entity(entity_type=legal_type, identifier=identifier)
+            if legal_type is LegalEntity.EntityTypes.SOLE_PROP.value:
+                alternate_name = factory_alternate_name(
+                    identifier=identifier,
+                    name="SP TESTING1212",
+                    name_type=AlternateName.NameType.DBA,
+                    bn15="111111100BC1111",
+                    start_date=datetime.utcnow(),
+                    legal_entity_id=legal_entity.id,
+                )
+                alternate_name.save()
+                legal_entity.alternate_names.append(alternate_name)
         else:
-            legal_entity = create_business(
-                entity_type=legal_type,
-                identifier=identifier,
-                create_office=True,
-                create_office_mailing_address=True,
-                create_office_delivery_address=True,
-                firm_num_persons_roles=2,
-                create_firm_party_address=True,
-                filing_types=["registration"],
-                filing_has_completing_party=[True],
-                create_completing_party_address=[True],
-            )
-        legal_entity.start_date = datetime.utcnow().date()
-        legal_entity.save()
-        session.commit()
+            if legal_type is LegalEntity.EntityTypes.SOLE_PROP.value:
+                legal_entity = factory_legal_entity(entity_type=owner_legal_type, identifier=owner_identifier)
+                alternate_name = create_alternate_name_business(
+                    entity_type=legal_type,
+                    identifier=identifier,
+                    create_office=True,
+                    create_office_mailing_address=True,
+                    create_office_delivery_address=True,
+                    firm_num_persons_roles=1,
+                    create_firm_party_address=True,
+                    filing_types=["registration"],
+                    filing_has_completing_party=[True],
+                    create_completing_party_address=[True],
+                )
+                mailing_address_alt = factory_address("mailing")
+                alternate_name.owner_mailing_address = mailing_address_alt
```

Comment:
if you are trying to test SP person scenario, then the mailing address should belong to the owning person LE record